lib/spack/spack/environment/environment.py

```diff
@@ -2933,95 +2933,43 @@ def included_config_scopes(self) -> List[spack.config.ConfigScope]:
             SpackEnvironmentError: if the manifest includes a remote file but
                 no configuration stage directory has been identified
         """
-        scopes: List[spack.config.ConfigScope] = []
-
-        # load config scopes added via 'include:', in reverse so that
-        # highest-precedence scopes are last.
         includes = self[TOP_LEVEL_KEY].get("include", [])
```

Comment:
>  I think this should use spack.config.get now, right? Otherwise it may not get the full list.

This processes the includes explicitly listed in the environment's manifest. The environment file has to be read before those includes are processes.